components/identities/actions/autoRenewalModal.tsx

```diff
@@ -0,0 +1,309 @@
+    address: etherContract?.address as string,
+    abi: etherContract?.abi as Abi,
+    functionName: "allowance",
+    args: [
+      address as string,
+      process.env.NEXT_PUBLIC_RENEWAL_CONTRACT as string,
+    ],
+  });
+  const { writeAsync: execute, data: autorenewData } = useContractWrite({
+    calls: callData,
+  });
+
+  useEffect(() => {
+    if (priceError || !priceData) setPrice("0");
+    else {
+      const high = priceData?.["price"].high << BigInt(128);
+      setPrice((priceData?.["price"].low + high).toString(10));
+    }
+  }, [priceData, priceError]);
+
+  useEffect(() => {
+    if (limitPrice === "0") setLimitPrice(price);
+  }, [address, domain, price, limitPrice]);
+
+  // on first load, we generate a salt
+  useEffect(() => {
+    setSalt(generateSalt());
+  }, []);
+
+  // we update compute the purchase metadata hash
+  useEffect(() => {
+    // salt must not be empty to preserve privacy
+    if (!salt) return;
+    (async () => {
+      setMetadataHash(
+        await computeMetadataHash(
+          email,
+          groups,
+          isUsResident ? usState : "none",
+          salt
+        )
+      );
+    })();
+  }, [usState, salt, email]);
+
+  useEffect(() => {
+    // check approval has been granted to renewal contract
+    if (approvalError || !approvalData) return;
+    if (
+      approvalData?.["remaining"].low === BigInt(0) &&
+      approvalData?.["remaining"].high === BigInt(0)
+    )
+      setNeedApproval(true);
+  }, [approvalData, approvalError]);
+
+  useEffect(() => {
+    if (isUsResident) {
+      salesTax.getSalesTax("US", usState).then((tax) => {
+        setSalesTaxRate(tax.rate);
+        if (price) setSalesTaxAmount(applyRateToBigInt(price, tax.rate));
+      });
+    } else {
+      setSalesTaxRate(0);
+    }
+  }, [isUsResident, usState, price]);
+
+  // Set Renewal Multicall
+  useEffect(() => {
+    if (isEnabled) {
+      setCallData(
+        registerCalls.disableRenewal(
+          callDataEncodedDomain[1].toString(),
+          limitPrice
+        )
+      );
+    } else {
+      const txMetadataHash = "0x" + metadataHash;
+      const finalPrice = Number(price) + Number(salesTaxAmount);
+      setCallData(
+        registerCalls.enableRenewal(
+          callDataEncodedDomain[1].toString(),
+          finalPrice.toString(),
+          txMetadataHash
+        )
+      );
+    }
+  }, [needApproval, isEnabled, price, salesTaxRate, metadataHash]);
+
+  useEffect(() => {
+    if (!autorenewData?.transaction_hash || !salt) return;
+    // posthog?.capture("register");
+
+    if (!isEnabled) {
+      // register the metadata to the sales manager db
+      // only when enabling auto renewal
+      fetch(`${process.env.NEXT_PUBLIC_SALES_SERVER_LINK}/add_metadata`, {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({
+          meta_hash: metadataHash,
+          email,
+          groups,
```

Comment:
We might add it several times to the same group, I think we have to rethink that with @Th0rgal

Reply:
Still problematic I think I'll make a new group

Reply:
Checked on DM